### PR TITLE
app/vmauth: allow dropping host header

### LINKED ISSUE #6453: vmauth: a way to stop proxying Host header
### Is your feature request related to a problem? Please describe

I tried to put VM stack into istio.
Istio is a service mesh that's intended for HTTP traffic mostly. It relies on `Host` header in HTTP requests to figure out what traffic policy/routing/etc to apply.

As of v1.101.0 version vmauth proxies `Host` header received from downstream to upstream services as-is.
This breaks istio because `Host` header vmauth sends to upstream is not upstream fqdn/dns name, but downstream entrypoint fqdn instead.

### Describe the solution you'd like

There are configuration options for deleting and adding headers on vmauth side. But `Host` header is a special case. Looking at the vmauth code I see that this header is protected when proxying request to upstream.
The only case when it's changed is TLS usage.

It would be nice to have a command line option, for example, that would tell vmauth to stop proxying `Host` header and use the upstream fqdn in this header instead.

### Describe alternatives you've considered

The only thing that worked for me was to tell istio to handle the traffic from vmauth to upstream svc as plain tcp (i. e. not http, so 90% of istio features become unavailable):
- create a separate k8s svc for upstream svc that has `tcp` word in ports name (or with appProtocol set to tcp)
- reconfigure vmauth to use this new svc as upstream

### Additional information

Istio is just an example. I believe other service mesh may be affected as well.
Other thing that comes to my mind is Autopilot service available in Google Cloud Platform (GCP).
I've never used it to be honest, but as far as I know they use istio under the hood.

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
### Describe Your Changes

Fixes #6453

### Checklist

The following checks are **mandatory**:

- [ ] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

File: app/vmauth/auth_config.go [hunk 1]
```diff
@@ -149,6 +150,15 @@ func (h *Header) MarshalYAML() (interface{}, error) {
 	return h.sOriginal, nil
 }
 
+func shouldDropHostHeader(headers []*Header) bool {
```
Comment (f41gh7) [inline app/vmauth/auth_config.go:153]: Naming looks a bit confusing to me. I'd expect from `dropHostHeader` option to completely remove host header from request and leave it empty.

 Maybe it's better to use - overrideHostHeader or enforceHostHeader WDYT?

Comment (AndrewChubatiuk) [reply app/vmauth/auth_config.go]: updated

**Comment by f41gh7** [comment]
Also a side question, our doc has the following statement: 
`# extra headers to add to the request or remove from the request (if header value is empty)`

Currently, for Host header it's only possible to set it to the target upstream (if value is empty). WDYT is it needed to replace it, if Value for Host header is set and not empty?

**Comment by AndrewChubatiuk** [comment]
> is it needed to replace it, if Value for Host header is set and not empty?
Setting custom static header value for host-based routing on a backend? I would rather wait for valid cases from users before adding this functionality

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by f41gh7** [comment]
Thanks for contribution!

**Comment by valyala** [comment]
@AndrewChubatiuk , please see the follow-up commit 851998ed96669c2bd20e8fc7ff732b08f6c66ef1

**Pull request merged**